### PR TITLE
Support extracting token deposit info from txs

### PR DESCRIPTION
**Pull request by Lbqds**

### CODE REVIEW

File: packages/web3/src/exchange/exchange.ts [hunk 1]
```diff
@@ -67,7 +58,54 @@ export function getALPHDepositInfo(tx: Transaction): { targetAddress: Address; d
...
+    }
+  }
+  return inputAddresses
+}
+
+export function isTokenTransferTx(tx: Transaction): boolean {
+  return isTransferTx(tx) && checkTokenOutput(tx)
+}
+
+export function getTokenDepositInfo(tx: Transaction): {
```
Comment (polarker) [inline packages/web3/src/exchange/exchange.ts:81]: Does this generate information for ALPH if ALPH and tokens are sent in a single transaction?

File: packages/web3/src/exchange/exchange.ts [hunk 2]
```diff
@@ -62,6 +62,7 @@ function getInputAddresses(tx: Transaction): Address[] {
   const inputAddresses: Address[] = []
   for (const input of tx.unsigned.inputs) {
     try {
+      if (input.unlockScript === '03') continue // SameAsPrevious
```
Comment (polarker) [inline packages/web3/src/exchange/exchange.ts:65]: IIRC, we have a constant somewhere for `03`, should reuse that one.

File: packages/web3/src/exchange/exchange.ts (hunk 1 above)
Comment (Lbqds) [reply packages/web3/src/exchange/exchange.ts:81]: No, it only returns the token info.

File: packages/web3/src/exchange/exchange.ts (hunk 2 above)
Comment (Lbqds) [reply packages/web3/src/exchange/exchange.ts]: We didn't define encoded `SameAsPrevious` before, so I added a constant for it in the codec

File: packages/web3/src/exchange/exchange.ts [hunk 3]
```diff
@@ -74,36 +79,44 @@ function getInputAddresses(tx: Transaction): Address[] {
...
-  depositAmount: bigint
-}[] {
-  if (!isTokenTransferTx(tx)) return []
+export interface DepositInfo {
+  alph: BaseDepositInfo[]
+  tokens: TokenDepositInfo[]
+}
+
+export function getDepositInfo(tx: Transaction): DepositInfo {
+  if (!isTransferTx) return { alph: [], tokens: [] }
```
Comment (polarker) [inline packages/web3/src/exchange/exchange.ts:92]: It should be `isTransferTx(tx)`

**Pull request merged**